sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/ScriptTransformation.scala

```diff
@@ -178,95 +169,156 @@ case class ScriptTransformation(
+    val outInfo: TableDesc =
+      if (outputSerdeClass != "") {
+        getTableDescFromSerDe(outputSerdeClass, columns, columnTypes, outputSerdeProps,
+          outputRowFormat)
+      } else {
+        PlanUtils.getTableDesc(defaultSerdeClass, fieldSeparator.toString, columns.mkString(","),
+          columnTypesNames, false)
+      }
+    outInfo
   }
 
   def parseAttrs(attrs: Seq[Expression]): (Seq[String], Seq[DataType]) = {
-                                                
-    val columns = attrs.map {
-      case aref: AttributeReference => aref.name
-      case e: NamedExpression => e.name
-      case _ => null
+    val columns = attrs.zipWithIndex.map {
+      case (aref: AttributeReference, _) => aref.name
+      case (e: NamedExpression, _) => e.name
+      case (o, index) => "c_$index"
     }
  
-    val columnTypes = attrs.map {
-      case aref: AttributeReference => aref.dataType
-      case e: NamedExpression => e.dataType
-      case _ =>  null
-    }
+    val columnTypes = attrs.map(_.dataType)
 
     (columns, columnTypes)
   }
- 
-  def initSerDe(serdeClassName: String, columns: Seq[String],
-    columnTypes: Seq[DataType], serdeProps: Seq[(String, String)]): AbstractSerDe = {
 
-    val serde: AbstractSerDe = if (serdeClassName != "") {
-      val trimed_class = serdeClassName.split("'")(1)
-      Utils.classForName(trimed_class)
-        .newInstance.asInstanceOf[AbstractSerDe]
-    } else {
-      null
-    }
-
-    if (serde != null) {
-      val columnTypesNames = columnTypes.map(_.toTypeInfo.getTypeName()).mkString(",")
+  def getTableDescFromSerDe(
+      serdeClassName: String,
+      columns: Seq[String],
+      columnTypes: Seq[DataType],
+      serdeProps: Seq[(String, String)],
+      rowFormat: Seq[(String, String, String)]): TableDesc = {
+    val columnTypesNames = columnTypes.map(_.toTypeInfo.getTypeName()).mkString(",")
+    if (serdeClassName != "") {
+      val className = serdeClassName
+      val serdeClass: Class[_ <: Deserializer] = Utils.classForName(className)
+        .asInstanceOf[Class[_ <: Deserializer]]
 
+      val tblDesc: TableDesc = PlanUtils.getTableDesc(serdeClass, Utilities.tabCode.toString,
+        columns.mkString(","), columnTypesNames, false)
+ 
       var propsMap = serdeProps.map(kv => {
-        (kv._1.split("'")(1), kv._2.split("'")(1))
+        (kv._1, kv._2)
       }).toMap + (serdeConstants.LIST_COLUMNS -> columns.mkString(","))
       propsMap = propsMap + (serdeConstants.LIST_COLUMN_TYPES -> columnTypesNames)
-    
-      val properties = new Properties()
-      properties.putAll(propsMap)
-      serde.initialize(null, properties)
-    }
+      propsMap.map(kv => tblDesc.getProperties().setProperty(kv._1, kv._2))
 
-    serde
+      return tblDesc
+    } else if (rowFormat != Nil) {
+      val tblDesc: TableDesc = PlanUtils.getDefaultTableDesc(Utilities.ctrlaCode.toString,
+        columns.mkString(","), columnTypesNames, false)
+      rowFormat.foreach { (tuple) =>
```

Comment:
Actually it doesn't keep ASTs but only the names of format nodes. Parsing is done in Hive parser. It is just string properties here.